Mage/src/main/java/mage/Mana.java

```diff
@@ -655,168 +754,168 @@ public boolean enough(final Mana cost) {
+            remaining += Math.min(0, compare.blue.getAmount());
+            remaining += Math.min(0, compare.black.getAmount());
+            remaining += Math.min(0, compare.red.getAmount());
+            remaining += Math.min(0, compare.green.getAmount());
+            remaining += Math.min(0, compare.colorless.getAmount());
+            remaining += Math.min(0, compare.any.getAmount());
             if (remaining > 0) {
-                int diff = Math.min(remaining, Math.abs(compare.generic));
-                compare.generic = compare.generic + diff;
+                int diff = Math.min(remaining, Math.abs(compare.generic.getAmount()));
+                compare.generic.incrementAmount(diff, false);
             }
         }
         Mana needed = new Mana();
-        if (compare.red < 0) {
-            needed.red = Math.abs(compare.red);
+        if (compare.white.getAmount() < 0) {
+            needed.white.removeAmount(compare.white);
         }
-        if (compare.white < 0) {
-            needed.white = Math.abs(compare.white);
+        if (compare.blue.getAmount() < 0) {
+            needed.blue.removeAmount(compare.blue);
         }
-        if (compare.green < 0) {
-            needed.green = Math.abs(compare.green);
+        if (compare.black.getAmount() < 0) {
+            needed.black.removeAmount(compare.black);
         }
-        if (compare.black < 0) {
-            needed.black = Math.abs(compare.black);
+        if (compare.red.getAmount() < 0) {
+            needed.red.removeAmount(compare.red);
         }
-        if (compare.blue < 0) {
-            needed.blue = Math.abs(compare.blue);
+        if (compare.green.getAmount() < 0) {
+            needed.green.removeAmount(compare.green);
         }
-        if (compare.colorless < 0) {
-            needed.colorless = Math.abs(compare.colorless);
+        if (compare.colorless.getAmount() < 0) {
+            needed.colorless.removeAmount(compare.colorless);
         }
-        if (compare.generic < 0) {
-            needed.generic = Math.abs(compare.generic);
+        if (compare.generic.getAmount() < 0) {
+            needed.generic.removeAmount(compare.generic);
         }
         return needed;
     }
 
     /**
-     * Returns total Red mana.
+     * Returns total White mana.
      *
-     * @return total Red mana.
+     * @return total White mana.
      */
-    public int getRed() {
-        return red;
+    public int getWhite() {
+        return white.getAmount();
     }
 
     /**
-     * Sets the total Red mana. Can not be negative. Negative values will be
+     * Sets the total White mana. Can not be negative. Negative values will be
      * logged and set to 0.
      *
-     * @param red total Red mana.
+     * @param white total White mana.
      */
-    public void setRed(int red) {
-        this.red = notNegative(red, "Red");
+    public void setWhite(int white) {
+        this.white.amount = notNegative(white, "White");
```

Comment:
You must clear snow stats on new values (call a set value method instead a direct -- like increment method do before). Snow stats can be potentially re-synced after set operation.